### PR TITLE
refactor: map to failure errors in X-Weave-Timestamp middleware

### LINKED ISSUE #38: Use failure crate everywhere
`api/handlers.rs` and `api/auth.rs` do their error-marshalling manually. The db code uses `failure` and I know it has magic in it for the marshalling stuff, I just never got round to reading about it yet.

### PR DESCRIPTION
**Pull request by philbooth**
Fixes #38 (I think?)

I was not feeling inspired when naming the new error type so ended up having a minor collision with actix's `ResponseError` trait. If one of you has a better name suggestion that avoids the need to alias that, let me know.

@bbangert @pjenvey r?

### CODE REVIEW

**Comment by pjenvey** [comment]
I consider these 3 more of a blanket "internal server error" as they're the result parsing data we were responsible for putting in the outgoing headers (so they should never happen).

I like having a blanket InternalServer error that takes a string for these. In the rare case it happens we can uniquely identify it (in a log or even in the outgoing error response) -- being a string its uniqueness isn't enforced by the type system -- but good enough for something that should never happen anyway *fingers crossed*

(a lot like what was already there, but we could have our own version of actix's internal server error or at the very least a custom json response for it if that's possible)

**Comment by philbooth** [comment]
> I like having a blanket InternalServer error that takes a string for these.

Understood, I can do that...

**Comment by philbooth** [comment]
> Understood, I can do that...

🧙

File: src/web/middleware.rs [hunk 1]
```diff
@@ -43,9 +43,20 @@ impl<S> Middleware<S> for WeaveTimestamp {
...
             let resp_ts = val
                 .to_str()
-                .map_err(|_| ErrorInternalServerError("Unable to extract header value"))?
-                .parse::<f64>()
-                .map_err(|_| ErrorInternalServerError("Invalid last-modified set"))?;
+                .map_err(|e| {
+                    let error: ApiError = ApiErrorKind::Internal(format!(
+                        "Invalid X-Last-Modified response header: {}",
+                        e
+                    )).into();
```
Comment (bbangert) [inline src/web/middleware.rs]: I kind of prefer this over turbofishing it.

**Comment by pjenvey** [comment]
Sorry for the lag I didn't see the last update

**Pull request merged**